### DISCUSSION TITLE
Hanging on delete?

### DISCUSSION THREAD

**Discussion by tnvu** [emmercm/igir#981] [Q&A] [answered]
Does anyone else have a problem where it seems like igir is hung when deleting files following a move command?

✓ Deleting moved files ······· | 2,544 moved files deleted

Has been sitting here for at least 2 hours. It's using a little CPU and some memory (according to top). 

When I strace/ltrace it is looping on some epoll_wait(13, ...). Looking a fd 13 it is "anon_inode:[eventpoll]" which isn't too useful to me.

I looked at the code and it looks like the delete code waits for all other threads to finish (I'm not too familiar with JS/Node/etc).

**Comment by emmercm** [comment]
The `✓` in the log line means the operation is finished, so there is something else hanging the process.

Please create a complete [bug report](https://github.com/emmercm/igir/issues/new/choose), including trace logs.

**Comment by tnvu** [comment]
I think this is related to #991. In Zip.CreateArchive if there's an error, it catches it and tries to delete the temporary zip file. I can still see the temporary zip file in my filesystem so it's definitely not being deleted.

**Reply by emmercm** [reply]
I think you are likely right. I suspect this promise here is never resolving: https://github.com/emmercm/igir/blob/5509c7852a861eb175beafc8e50f9bdb10b43673/src/types/files/archives/zip.ts#L220-L228

**Reply by emmercm** [reply, marked as answer]
https://github.com/emmercm/igir/pull/994 should have prevented this specific issue.
